infra-cdk/lib/stacks/IAMStack.ts

```diff
@@ -17,18 +20,34 @@ export class IAMStack extends cdk.Stack {
     constructor(scope: Construct, id: string, props: IAMStackProps) {
         super(scope, id, props);
 
+         /** Setted up the option to choose between S3 vs DynamoDB or both */
```

Comment:
This is a nice way to set it up if they are optional! I would recommend making all the props mandatory, then this refactor won't be necessary